roiextractors/imagingextractor.py

```diff
@@ -58,6 +58,38 @@ def get_dtype(self) -> DtypeType:
     def get_video(self, start_frame: int = None, end_frame: int = None, channel: int = 0) -> NumpyArray:
         return self.get_frames(range(start_frame, end_frame), channel)
 
+    def frame_to_time(self, frame: int):
+        '''This function converts a user-inputted frame index to a time with units of seconds.
+
+        Parameters
+        ----------
+        frame: float
+            The frame to be converted to a time
+
+        Returns
+        -------
+        time: float
+            The corresponding time in seconds
+        '''
+        # Default implementation
+        return frame / self.get_sampling_frequency()
+
+    def time_to_frame(self, time):
+        '''This function converts a user-inputted time (in seconds) to a frame index.
+
+        Parameters
+        -------
+        time: float
+            The time (in seconds) to be converted to frame index
+
+        Returns
+        -------
+        frame: float
+            The corresponding frame index
+        '''
+        # Default implementation
+        return time * self.get_sampling_frequency()
```

Comment:
handle non-zero start time

Reply:
this can be overwritten in derived-classes to handle non-zero start times